tests/unit/core/reducers/test_autocomplete.js

```diff
@@ -0,0 +1,90 @@
+import reducer, {
+  autocompleteAbort,
+  autocompleteLoad,
+  autocompleteStart,
+  AUTOCOMPLETE_STARTED,
+  AUTOCOMPLETE_LOADED,
+} from 'core/reducers/autocomplete';
+import { createFakeAutocompleteResult } from 'tests/unit/amo/helpers';
+
+describe(__filename, () => {
+  describe('reducer', () => {
+    it('initializes properly', () => {
+      const { loading, suggestions } = reducer(undefined);
+      expect(loading).toBe(false);
+      expect(suggestions).toEqual([]);
+    });
+
+    it('handles AUTOCOMPLETE_ABORTED', () => {
+      const { loading, suggestions } = reducer(undefined, autocompleteAbort());
+      expect(loading).toBe(false);
+      expect(suggestions).toEqual([]);
+    });
+
+    it('handles AUTOCOMPLETE_STARTED', () => {
+      const { loading, suggestions } = reducer(undefined, { type: AUTOCOMPLETE_STARTED });
+      expect(loading).toBe(true);
+      expect(suggestions).toEqual([]);
+    });
+
+    it('handles AUTOCOMPLETE_LOADED', () => {
+      const payload = {
+        results: [
+          createFakeAutocompleteResult({ name: 'foo' }),
+          createFakeAutocompleteResult({ name: 'bar' }),
+          createFakeAutocompleteResult({ name: 'baz' }),
+        ],
+      };
+
+      const { loading, suggestions } = reducer(undefined, {
+        type: AUTOCOMPLETE_LOADED,
+        payload,
+      });
+      expect(loading).toBe(false);
+      expect(suggestions).toEqual(['foo', 'bar', 'baz']);
+    });
+
+    it('excludes AUTOCOMPLETE_LOADED results with null names', () => {
+      const payload = {
+        results: [
+          createFakeAutocompleteResult({ name: 'foo' }),
+          createFakeAutocompleteResult({ name: null }),
+          createFakeAutocompleteResult({ name: 'baz' }),
+        ],
+      };
+
+      const { loading, suggestions } = reducer(undefined, {
+        type: AUTOCOMPLETE_LOADED,
+        payload,
+      });
+      expect(loading).toBe(false);
+      expect(suggestions).toEqual(['foo', 'baz']);
+    });
+  });
+
+  describe('autocompleteStart()', () => {
+    it('throws an error if arguments are missing', () => {
+      expect(() => autocompleteStart({})).toThrow(/errorHandlerId is required/);
```

Comment:
I feel like we've been out of the habit of using implicit returns. I get they're shorter but I find them a fair bit harder to read, especially here in the test where we aren't focused on a function's *return value* but instead its *side effect* (throwing the exception). I feel like this test would be easier to immediately grok as:

```js
expect(() => {
  autocompleteStart({});
}).toThrow(/errorHanderId is required/);
```

I know it's more to write but I think it gives the code room to breathe (an aesthetic preference I'll admit) but also more accurately represents what the test is about.

I'd prefer no implicit returns because I don't like concise syntax like that, but I think they're really only useful when they fit on one line and are actually intended to return something.